### DISCUSSION TITLE
How does the cox predict work exactly?

### DISCUSSION THREAD

**Discussion by justkrismanohar** [CamDavidsonPilon/lifelines#1584] [Q&A]
HI guys,

So, from my understanding, the Cox model does not assume any form of the base link hazard function h0. So, I am wording how lifelines give us probabilities for survival. Is it assuming form for h0? I am really interested to understand how this works as I am new to survival analysis.

Thanks guys!

**Comment by CamDavidsonPilon** [comment]
You can compute the baseline hazard non-parametrically (not unlike how the Kaplan Meier curve is computed non-parametrically, but the two are not the same). Given h0 and the hazard coefficients, you have the full survival curve => predictions.

**Reply by justkrismanohar** [reply]
Interesting..... i noticed in https://github.com/CamDavidsonPilon/lifelines/issues/543 you mentioned that 
> > In Cox PH Regression model, the baseline hazard function is the hazard function when all covariates are set to zero, right?
> 
> So, no. In lifelines, the baseline hazard is the hazard function when all covariates are set _to the mean of the covariate_.
> 
> So that "baseline survival" in the graph is the "average" subject (which often has no real world meaning tbh, you can't be 0.5 male). Thus, it makes sense to have prio=0 above the baseline survival.
> 
> Does that make sense?

I assume in that case the baseline will take on a form? so something like e^(Beta*X) ? But I think I am missing something and the h0 should be function of t?

**Reply by CamDavidsonPilon** [reply]
well hazard is `h(t) = h0(t) * exp(something)`. That conversation is about what `something` is s.t. `exp(something)` is equal to 1, which provides an _interpretation_ to `h0(t)`. However `h0(t)` is still computed non-parametrically.
